app/scripts/components/globalComponents/metadataTemplateList.jsx

```diff
@@ -0,0 +1,264 @@
+import React from 'react';
+import { RouteHandler } from 'react-router';
+import MainActions from '../../actions/mainActions';
+import ProjectActions from '../../actions/projectActions';
+import ProjectStore from '../../stores/projectStore';
+import AddAgentModal from '../../components/globalComponents/addAgentModal.jsx';
+import Dialog from 'material-ui/lib/dialog';
+import FlatButton from 'material-ui/lib/flat-button';
+import Help from 'material-ui/lib/svg-icons/action/help';
+import IconButton from 'material-ui/lib/icon-button';
+import Loaders from '../../components/globalComponents/loaders.jsx';
+import Paper from 'material-ui/lib/paper';
+import urlGen from '../../../util/urlGen.js';
+import RaisedButton from 'material-ui/lib/raised-button';
+import Search from 'material-ui/lib/svg-icons/action/search';
+import Close from 'material-ui/lib/svg-icons/navigation/close';
+import TextField from 'material-ui/lib/text-field';
+import Toggle from 'material-ui/lib/toggle';
+
+class MetadataTemplateList extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            searchMode: false,
+            timeout: null,
+            toggleSwitch: false
+        };
+    }
+
+    render() {
+        if (this.props.error && this.props.error.response) {
+            this.props.error.response === 404 ? this.props.appRouter.transitionTo('/notFound') : null;
+            this.props.error.response != 404 ? console.log(this.props.error.msg) : null;
+        }
+        let currentUser = this.props.currentUser && this.props.currentUser !== null ? this.props.currentUser : null;
```

Comment:
sometimes it's nice to create local variables for props for code clarity. It's easier with object destructuring. ex:

```
const { error, currentUser } = this.props;
```